src/main/java/hudson/plugins/gradle/injection/BuildScanInjection.java

```diff
@@ -2,14 +2,33 @@
 
 import hudson.EnvVars;
 import hudson.model.Node;
+import hudson.model.labels.LabelAtom;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public interface BuildScanInjection {
 
-    default boolean isOn(EnvVars env) {
+    default boolean injectionEnabled(EnvVars env) {
         return EnvUtil.isSet(env, getActivationEnvironmentVariableName());
     }
 
     String getActivationEnvironmentVariableName();
 
     void inject(Node node, EnvVars envGlobal, EnvVars envComputer);
+
+    default boolean isInjectionEnabledForNode(Node node, EnvVars envGlobal) {
+        Set<LabelAtom> labelAtoms = (node.getAssignedLabels() != null ? node.getAssignedLabels() : new HashSet<>());
+        Set<String> labels = labelAtoms.stream().map(LabelAtom::getName).collect(Collectors.toSet());
```

Comment:
Probably it's better to join these 2 lines. Intermediate variable here doesn't bring much value I think: 

```suggestion
        Set<String> labels =
            (node.getAssignedLabels() != null ? node.getAssignedLabels() : Collections.<LabelAtom>emptySet())
                .stream()
                .map(LabelAtom::getName)
                .collect(Collectors.toSet());
```

Reply:
Done